liferay-echo/liferay-portal
# LPS-200275 FragmentEntryVersion always has draft status

### Motivation
https://liferay.atlassian.net/browse/LPS-200275

We don't allow publishing a publication with draft changes. However, when a user adds and publishes a new fragment, its FragmentEntryVersion always has the status set to `draft` and there's no way to update it to approved. The issue is when the fragmentEntry is published, the FragmentEntryVersion is populated with the values from the draft entry (https://github.com/liferay/liferay-portal/blob/master/modules/apps/fragment/fragment-service/src/main/java/com/liferay/fragment/service/base/FragmentEntryLocalServiceBaseImpl.java#L867). And even though the [status is set to approved](https://github.com/liferay/liferay-portal/blob/master/modules/apps/fragment/fragment-service/src/main/java/com/liferay/fragment/service/base/FragmentEntryLocalServiceBaseImpl.java#L867) before the publish, this value is ignored because of the [getStatus override](https://github.com/liferay/liferay-portal/blob/master/modules/apps/fragment/fragment-service/src/main/java/com/liferay/fragment/model/impl/FragmentEntryImpl.java#L97-L103).

### Proposed Solution
Ideally, we would not hard-code the status since it's used to create the model versions. However, we don't understand the use cases. So the fix we proposed is to use the status value set on the object when creating the version object.

### Steps to Verify
1. Add a fragment and publish
2. Verify in the database that FragmentEntryVersion has draft status = 2

Thank you!
